dashboard/app/helpers/levels_helper.rb

```diff
@@ -182,6 +182,17 @@ def app_options
       # currently, all levels are Blockly or DSLDefined except for Unplugged
       @app_options = view_options.camelize_keys
     end
+
+    @app_options[:dialog] = {
+      skipSound: !!(@level.properties['options'].try(:[], 'skip_sound')),
+      preTitle: @level.properties['pre_title'],
+      fallbackResponse: @fallback_response.to_json,
+      callback: @callback,
+      app: @level.type.underscore,
```

Comment:
Ok. so there's another set of app names defined in the various <app>.js files under apps/src/... when they fire their own milestone posts, that's where turtle, Jigsaw, and most of the rest of the names actually come from.

The app name I'm setting above is only actually used for milestone posts coming from external, multi, match, free response, and levelgroup levels.